### ISSUE TITLE
More flexibility in requirements.txt format

### ISSUE THREAD

**Issue by brondsem** [twu/skjold#3]
If a requirements file has anything extra, this error is raised: 
```
  File "/Users/brondsem/tmp/py3venv/lib/python3.6/site-packages/skjold/cli.py", line 169, in audit_
    packages = extract_package_list_from(config, file, file_format)
  File "/Users/brondsem/tmp/py3venv/lib/python3.6/site-packages/skjold/formats.py", line 80, in extract_package_list_from
    for package in reader_func(file):
  File "/Users/brondsem/tmp/py3venv/lib/python3.6/site-packages/skjold/formats.py", line 47, in read_requirements_txt_from
    package_name, package_version = line.strip().split(" ")[0].split("==")
ValueError: not enough values to unpack (expected 2, got 1)
```

The most common example that can cause this is `#` for comments.

I also have requirements.txt files with hashes specified and spanning multiple lines, like this:
```
beautifulsoup4==4.8.0 \
    --hash=sha256:05668158c7b85b791c5abde53e50265e16f98ad601c402ba44d70f96c4159612 \
    --hash=sha256:25288c9e176f354bf277c0a10aa96c782a6a18a17122dba2e8cec4a97e03343b \
    --hash=sha256:f040590be10520f2ea4c2ae8c3dae441c7cfff5308ec9d58a0ec0c1b8f81d469
```

There are other things that can be in a valid requiremens.txt file too, like extra pip options.  https://pip.pypa.io/en/latest/reference/pip_install/#requirements-file-format

These can all be worked around by massaging the requirements file first and passing it as stdin something like: `cat requirements.txt | sed 's/#.*//' | skjold audit -s gemnasium -`  But it would be very handy for skjold to handle any requirements.txt file that is valid for pip

**Comment by twu** [comment]
Hej!

What specific  _extras_ do you have in mind?
Can you provide a set of _extras_ that you would need support for?
Maybe even provide a few example `requirements.txt` I can test against?

Hashes shouldn't be a problem (I also just added tests to prove that). I also added a small workaround for comments or lines starting with  `#` (https://github.com/twu/skjold/commit/332ca41fa4985a42e753aeb3a0428e545d29be5f).

**Background**:
The current "support" to read a `requirements.txt`-like format is very, very limited and only comes from the fact that the _pinned_ or _frozen_ format (?) (i.e. `package==0.1.2`) is used to be able to integrate with other tools or commands such as `pip freeze`, `poetry export -f requirements.txt` etc.

Skjold should be able to handle this very basic _frozen/pinned_ `requirements.txt`-like formatted input (from e.g. `pip freeze`) and also has some (simple/stupid) workarounds for stuff I encountered e.g. `; platform=` and `--hashes`  when using those tools.

Fully supporting the `requirements.txt` format in my head sounds like also having to perform dependency resolution in one way or the other - which IMHO - I would like to leave to better tools.

What do you think?

**Comment by brondsem** [comment]
The extras in my requirements file are like this:
```
--extra-index-url http://internal-host/pypi/index/
--trusted-host internal-host
```

I've just tested your other recent changes and they're working great.

I think your overall direction of handling only pinned/frozen requirements makes sense.  I didn't mean to say skjold should support all the semantics of any requirements file, just the syntax, so if there's a line like `--extra-index-url http://whatever/` to ignore it and not error. 

I don't have any unpinned packages (I'm using `pip-compile` to generate my requirements file) but it might be nice to handle lines like `requests` or `requests > 2` (or anything else that is unsupported/unparsed) by emitting a warning.  Then if someone tries skjold on a file with those, they get a message telling them which line is ignored, rather than an exception.

**Comment by twu** [comment]
> I didn't mean to say skjold should support all the semantics of any requirements file, just the syntax, so if there's a line like `--extra-index-url http://whatever/` to ignore it and not error.

Oh, I see. Thanks for the clarification :) Makes total sense. Will try to add it in the next few days.

**Comment by twu** [comment]
Implemented a fix with ca40e1fd2839712b4d4e23d2d3b41d78fd17170e that got released with `v0.1.5`.
